frontend/packages/operator-lifecycle-manager/src/components/operand.tsx

```diff
@@ -167,8 +167,54 @@ export const OperandTableHeader = () => {
   ];
 };
 
+export enum OperatorStatusType {
+  phase = 'phase',
+  status = 'status',
+  state = 'state',
+  conditions = 'conditions',
+}
+
+export const OperatorStatusTypeText = {
+  [OperatorStatusType.phase]: 'Phase:',
+  [OperatorStatusType.status]: 'Status:',
+  [OperatorStatusType.state]: 'State:',
+  [OperatorStatusType.conditions]: 'Condition:',
+};
+
+export const OperandStatusIconAndText: React.FunctionComponent<OperandStatusIconAndTextProps> = ({
+  statusObject,
+}) => {
+  let iconAndText = <div className="text-muted">Unknown</div>;
+  if (_.isEmpty(statusObject)) {
+    return iconAndText;
+  }
+  _.each(OperatorStatusType, (key: string) => {
+    if (_.has(statusObject, key)) {
+      const status = statusObject[key];
+      if (!_.isEmpty(status)) {
+        let statusIcon;
+        if (key !== OperatorStatusType.conditions) {
+          statusIcon =
+            status === 'Running' ? <SuccessStatus title={status} /> : <Status status={status} />;
+        } else if (_.has(status[0], 'type')) {
+          statusIcon = <Status status={status[0].type} />;
+        }
+        if (statusIcon) {
+          iconAndText = (
+            <span className="co-icon-and-text">
+              {OperatorStatusTypeText[key]}&nbsp;{statusIcon}
+            </span>
+          );
+          return false;
```

Comment:
I think there may be a better util to use than `_.each` with the `return true||false` as a short circuit.  Have you considered a `_.find` or a `_.filter`?

Reply:
used _.find instead. thanks for the comment.